cpp/src/io/json/json_tree.cu

```diff
@@ -632,23 +621,27 @@ std::pair<rmm::device_uvector<size_type>, rmm::device_uvector<size_type>> hash_n
     return node_id1 == node_id2;
   };
 
+  constexpr size_type empty_node_index_sentinel = -1;
+  using hash_table_allocator_type = rmm::mr::stream_allocator_adaptor<default_allocator<char>>;
+  using hasher_type               = decltype(d_hashed_cache);
+
+  auto key_set = cuco::experimental::static_set{
+    cuco::experimental::extent{compute_hash_table_size(num_nodes)},
+    cuco::empty_key<cudf::size_type>{empty_node_index_sentinel},
+    d_equal,
+    cuco::experimental::linear_probing<1, hasher_type>{d_hashed_cache},
```

Comment:
Why using `linear_probing` here? Do we have other options?

Reply:
used the same probing used in distinct_count. What other probing options are there?

Reply:
@PointKernel?

Reply:
We can switch between `linear_probing` and `double_hashing`. In general, `linear_probing` delivers better performance for set/map and `double_hashing` is preferred for multiset/multiset.